### PR TITLE
Added tests for input/output filenames for ICON and EMAC on-the-fly CMORizer

### PR DESCRIPTION
**Pull request by schlunma**
## Description

While coding for #1678 I realized that we currently don't have any tests for input file names/input directories/output files for the EMAC and ICON on-the-fly CMORizers. This PR adds those. In additon, I slightly adapted the output file name for EMAC to make it consistent with CMIP data.

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Comment by schlunma** [comment]
@valeriupredoi would you have time to briefly look at this? Thanks so much!!

**Comment by valeriupredoi** [comment]
@schlunma cheers, yeps - will do today, in a jiffy!

**Review by valeriupredoi — APPROVED** [review]
looking good, cheers Manu! One question related to the native file names (sorry, first time I sees them), and another question - do we have true negative tests? ie a group of files that *should not* be found - I can't remember if I implemented that initially, so if there are none for CMIP data that's fine, we can leave it like that

File: tests/integration/data_finder.yml [hunk 1]
```diff
@@ -779,3 +850,141 @@ get_input_filelist:
...
+      project: EMAC
+      cmor_table: CMIP6
+      frequency: mon
+      mip: Amon
+      exp: amip
+      timerange: '200002/200003'
+      diagnostic: test_diag
+      preprocessor: test_preproc
+    available_files:
+      - amip/Amon/amip___________200001_Amon.nc
```
[the remark this answers is not in the document]
Comment (schlunma) [reply tests/integration/data_finder.yml:871]: Yes, that's exactly how they look :smile: The number of underscores depends on the length of the `exp`; longer experiments have less underscores and vice versa

```bash
Amon                                                                                                                                                                                                                                                                          
├── RD2-base-01a___195001_Amon.nc                                                                                                                                                                                                                                             
├── RD2-base-01a___195002_Amon.nc                                                                                                                                                                                                                                             
├── RD2-base-01a___195003_Amon.nc                                                                                                                                                                                                                                             
├── RD2-base-01a___195004_Amon.nc
...
```

**Comment by schlunma** [comment]
This also has negative tests:

```yml
available_files:
      - amip/Amon/amip___________200001_Amon.nc
      - amip/Amon/amip___________200002_Amon.nc
      - amip/Amon/amip___________200003_Amon.nc
      - amip/Amon/amip___________200001_Amon-p-mm.nc
      - amip/Amon/amip___________200002_Amon-p-mm.nc
      - amip/Amon/amip___________200003_Amon-p-mm.nc
      - amip/rad/amip___________200001_rad.nc
      - amip/rad/amip___________200002_rad.nc
      - amip/rad/amip___________200003_rad.nc
      - amip/rad/amip___________200001_rad-p-mm.nc
      - amip/rad/amip___________200002_rad-p-mm.nc
      - amip/rad/amip___________200003_rad-p-mm.nc
    dirs:
      - amip/Amon
    file_patterns:
      - amip*Amon.nc
    found_files:
      - amip/Amon/amip___________200002_Amon.nc
      - amip/Amon/amip___________200003_Amon.nc
```

As you can see, only a subset of the files available is supposed to be found.

File: tests/integration/data_finder.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (valeriupredoi) [reply tests/integration/data_finder.yml:871]: cool, thanks, man! I mean not cool - that's such a silly way of naming files but heyho :laughing:

**Comment by valeriupredoi** [comment]
> This also has negative tests:
> 
> ```yaml
> available_files:
>       - amip/Amon/amip___________200001_Amon.nc
>       - amip/Amon/amip___________200002_Amon.nc
>       - amip/Amon/amip___________200003_Amon.nc
>       - amip/Amon/amip___________200001_Amon-p-mm.nc
>       - amip/Amon/amip___________200002_Amon-p-mm.nc
>       - amip/Amon/amip___________200003_Amon-p-mm.nc
>       - amip/rad/amip___________200001_rad.nc
>       - amip/rad/amip___________200002_rad.nc
>       - amip/rad/amip___________200003_rad.nc
>       - amip/rad/amip___________200001_rad-p-mm.nc
>       - amip/rad/amip___________200002_rad-p-mm.nc
>       - amip/rad/amip___________200003_rad-p-mm.nc
>     dirs:
>       - amip/Amon
>     file_patterns:
>       - amip*Amon.nc
>     found_files:
>       - amip/Amon/amip___________200002_Amon.nc
>       - amip/Amon/amip___________200003_Amon.nc
> ```
> 
> As you can see, only a subset of the files available is supposed to be found.

gotcha! Too Monday morning for me to catch that - good stuff! All fine by me then, cheers for the good work  bud! :beer:

**Comment by schlunma** [comment]
Thanks for the quick review :tada:

**Pull request merged**